### ISSUE TITLE
Mailbox is not member of any allowed group 

### ISSUE THREAD

**Issue by sjteoh111** [GruberMarkus/Set-OutlookSignatures#82]
<img width="499" alt="image" src="https://github.com/GruberMarkus/Set-OutlookSignatures/assets/115065492/85170322-5922-46a4-b6ed-883b67267cb2">

i have the error above no legacyExchangeDN , is this error causing mailbox not member of allowed group ? i check my legacyExchangeDN is empty and my environment is Hybrid AD.

**Comment by GruberMarkus** [comment]
Hi @sjteoh111,

The warning that no legacyExchangeDn was found hints to one of the following scenarios:
- The mailbox is not an Exchange mailbox
- The Outlook profile is damaged
- Sync between on-prem and Azure AD/Entra ID is not configured correctly
- Set-OutlookSignatures is not configured correctly for your environment


As you write that you use hybrid AD. Do you still have Exchange servers on-prem, or sync Exchange attributes between your local Active Directory and Azure AD/Entra ID?

I guess that you do not have an Exchange server on-prem anymore (or never had Exchange on-prem), and therefore do not sync the corresponding attributes.

You can either sync the attributes to on-prem and let Set-OutlookSignatures access your local Active Directory, or you can force Set-OutlookSignatures to only use Azure AD/Entra ID/Graph.

This is described in the README file in the 'Quick Start Guide'. as well as in 'Hybrid and cloud-only support'.
In short:
- In Azure AD/Entra ID register the Set-OutlookSignatures application and grant delegated permissions, as described in '.\config\default graph config.ps1' and the chapters mentioned above
- Then, run Set-OutlookSignatures with the '-GraphOnly true' parameter

Please let me know if this helps.

**Comment by sjteoh111** [comment]
i have change -GraphOnly to True , but in _signature.ini **not for direct and indirect members of a group or distribution list** the group cannot be found . previously then -GraphOnly false able find the group sid .

**Comment by GruberMarkus** [comment]
Please provide a full log as text file (not as screenshot), created with the -verbose parameter.

**Comment by GruberMarkus** [comment]
Hi @sjteoh111,

if you could solve the problem yourself in the meantime, please let me and the other readers here know how you solved it. This will help others which are in a comparable situation in the future.

If the problem is unsolved, please provide a full log as text file (not as screenshot), created with the -verbose parameter, so I can start analyzing.

**Comment by GruberMarkus** [comment]
Closing the issue, as @sjteoh111 has not provided feedback for a week.

This issue can of course be re-opened when the problem appears again and log files can be provided.
